codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/auth/http/integration/AwsSdkCustomizeHttpBearerTokenAuth.java

```diff
@@ -55,7 +55,9 @@ public void customizeSupportedHttpAuthSchemes(
                     .addDependency(AwsDependency.TOKEN_PROVIDERS)
                     .addImport("nodeProvider", null, AwsDependency.TOKEN_PROVIDERS)
                     .addImport("FromSsoInit", null, AwsDependency.TOKEN_PROVIDERS)
-                    .write("async (idProps) => await nodeProvider(idProps as FromSsoInit)(idProps)"))
+                    .openBlock("async (idProps) => {", "}", () -> {
+                        w.write("return await nodeProvider(idProps as FromSsoInit)(idProps);");
```

Comment:
async function can return without await

Reply:
It can, but we want errors to be thrown inside the function block, for changes coming up in future.
Not specifically for this line, but when we'll add a try block in future.

If it returns without await the error will be thrown where it's awaited by callee.